flink/src/test/java/org/apache/iceberg/flink/TestFlinkSchemaUtil.java

```diff
@@ -270,4 +275,54 @@ private void checkInconsistentType(
         Types.StructType.of(Types.NestedField.optional(0, "f0", icebergExpectedType)),
         FlinkSchemaUtil.convert(FlinkSchemaUtil.toSchema(RowType.of(flinkType))).asStruct());
   }
+
+  @Test
+  public void testConvertFlinkSchemaBaseOnIcebergSchema() {
+    Schema baseSchema = new Schema(
+        Lists.newArrayList(
+            Types.NestedField.required(101, "int", Types.IntegerType.get()),
+            Types.NestedField.optional(102, "string", Types.StringType.get())
+        ),
+        Sets.newHashSet(101, 102)
+    );
+
+    TableSchema flinkSchema = TableSchema.builder()
+        .field("int", DataTypes.INT().notNull())
+        .field("string", DataTypes.STRING().nullable())
+        .primaryKey("int")
+        .build();
+    Schema convertedSchema = FlinkSchemaUtil.convert(baseSchema, flinkSchema);
+    Assert.assertEquals(baseSchema.asStruct(), convertedSchema.asStruct());
+    Assert.assertEquals(ImmutableSet.of(101), convertedSchema.identifierFieldIds());
+  }
+
+  @Test
+  public void testConvertFlinkSchemaWithPrimaryKeys() {
+    Schema iSchema = new Schema(
```

Comment:
nit: iSchema -> icebergSchema? It took me a few seconds to guess what `i` stands for here :)